### PR TITLE
Marking Obsolete SiteRecovery and classic SiteRecovery. RecoveryServices.SiteRecovery can be used instead.

### PR DESCRIPTION
**Pull request by viverm**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by viverm** [comment]
Need data for Next breaking release from where we can stop support for old modules.

File: src/ResourceManager/SiteRecovery/documentation/upcoming-breaking-changes.md [hunk 1]
```diff
@@ -25,4 +25,81 @@
...
+
+## Deprecation of the AzureRm.SiteRecovery module
+
+The AzureRm.SiteRecovery module will be deprecated in an upcoming release.
+
+The AzureRm.SiteRecovery module is being superseded by the AzureRm.RecoveryServices.SiteRecovery module.
+
+The AzureRm.RecoveryServices.SiteRecovery is a functional superset of the AzureRm.SiteRecovery module and includes a new set of equivalent cmdlets and easy to use aliases to help you manage your Azure Site Recovery deployments.
+
+|--------------------------------------------------------|---------------------------------------------------------------------------------------------|----------------------------------------------------|
```
Comment (cormacpayne) [inline src/ResourceManager/SiteRecovery/documentation/upcoming-breaking-changes.md]: @viverm the table is getting parsed incorrectly in markdown - removing the first and last row of this table fixed the issue for me

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Fabrics/GetAzureRmSiteRecoveryFabric.cs [hunk 2]
```diff
@@ -29,6 +29,9 @@ namespace Microsoft.Azure.Commands.SiteRecovery
     /// </summary>
     [Cmdlet(VerbsCommon.Get, "AzureRmSiteRecoveryFabric", DefaultParameterSetName = ASRParameterSets.Default)]
     [OutputType(typeof(List<ASRFabric>))]
+    [Obsolete("This cmdlet has been marked for deprecation in an upcoming release. Please use the " +
```
Comment (cormacpayne) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Fabrics/GetAzureRmSiteRecoveryFabric.cs]: @viverm for all of these `Obsolete` attributes, I would point at the `upcoming-breaking-changes.md` file with the cmdlet mapping table to make it easier for users to identify which cmdlets to use in the new module

**Comment by cormacpayne** [comment]
@viverm when pointing customers at the `upcoming-breaking-changes.md` file, you can point them [here](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/SiteRecovery/documentation/upcoming-breaking-changes.md), and once this PR gets merged, it will contain the migration table. Any issue with that?

**Comment by viverm** [comment]
@Cormac As discussed already added the replacement cmdlet in obsolete message .

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/229/

**Pull request merged**